src/rebloom.c

```diff
@@ -229,20 +230,28 @@ static int bfInsertCommon(RedisModuleCtx *ctx, RedisModuleString *keystr, RedisM
     }
 
     if (options->is_multi) {
-        RedisModule_ReplyWithArray(ctx, nitems);
+        RedisModule_ReplyWithArray(ctx,  REDISMODULE_POSTPONED_ARRAY_LEN);
     }
 
-    for (size_t ii = 0; ii < nitems; ++ii) {
+    size_t array_len = 0;
+    for (size_t ii = 0; ii < nitems && result == REDISMODULE_OK; ++ii) {
         size_t n;
         const char *s = RedisModule_StringPtrLen(items[ii], &n);
         int rv = SBChain_Add(sb, s, n);
         if (rv == -2) { // decide if to make into an error
-            return RedisModule_ReplyWithError(ctx, "Non scaling filter is full");
+            RedisModule_ReplyWithError(ctx, "Non scaling filter is full");
+            result = REDISMODULE_ERR;
```

Comment:
what are you doing with the error reply?
I think we should still return OK and the user will see the error in the array.

Reply:
@ashtul to keep the same behaviour as before, we need to return `REDISMODULE_ERR`

Reply:
`RedisModule_ReplyWithError` always returns `REDISMODULE_OK`

Reply:
you're totally right. fixed on the new commit